spring-context-support/src/main/java/org/springframework/cache/caffeine/CaffeineCacheManager.java

```diff
@@ -59,7 +60,9 @@
 
 	private boolean dynamic = true;
 
-	private Caffeine<Object, Object> cacheBuilder = Caffeine.newBuilder();
+	private final ConcurrentMap<String, Caffeine> cacheBuilder = new ConcurrentHashMap<>(16);
```

Comment:
I wonder if we shouldn't put that outside of the `CacheManager` business. After all if you want to configure your infrastructure, that could/should happen upfront. I think this would also vastly simplify the current arrangement where we have to manage the state if the setter is called once some caches are already being built.